command.js

```diff
@@ -0,0 +1,117 @@
+var exec = require('child_process').exec;
+var Promise = require('promise');
+var debug = require('debug')('mozdevice:command');
+
+/**
+ * API for building up and executing shell commands
+ * @param {string} [initialCommand] An optional first command
+ * @constructor
+ */
+var Command = function(initialCommand) {
+  this.builder = initialCommand ? [initialCommand] : [];
+};
+
+/**
+ * Append content to the command builder
+ * @param {string} content
+ * @returns {Command}
+ */
+Command.prototype.append = function(content) {
+  this.builder.push(content);
+  return this;
+};
+
+/**
+ * Append a command to be run against `adb <command>`
+ * @param {string} command
+ * @returns {Command}
+ */
+Command.prototype.adb = function(command) {
+  this.builder.push('adb ' + command);
+  return this;
+};
+
+/**
+ * Append a command to be run against `adb shell <command>`
+ * @param {string} command
+ * @returns {Command}
+ */
+Command.prototype.adbShell = function(command) {
+  this.builder.push("adb shell '" + command + "'");
```

Comment:
Just a question: Why do you need quotes around the actual 'command' for 'adb shell' but not for just the 'adb' <command> above? Using 'adb shell' with commands that are > 1 word works fine without quotes (at least on the command line).